Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -520,34 +542,46 @@ extension SwiftLanguageServer {
       // Clear settings that should not be cached for closed documents.
       self.commandsByFile[uri] = nil
       self.currentDiagnostics[uri] = nil
+      self.documentParseTransition[uri] = nil
+      self.documentReusedNodeCollector[uri] = nil
 
       _ = try? self.sourcekitd.sendSync(req)
     }
   }
 
   public func changeDocument(_ note: DidChangeTextDocumentNotification) {
```

Comment:
Have you checked whether the LSP edits are sequential or concurrent? I’m not sure if the spec defines it so you might need to just check what kind of edits VSCode sends when you do a multi-cursor edit.

Reply:
The [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#didChangeTextDocumentParams) indicates that the edits are sequential IIUC.